azure-mgmt-network/src/main/java/com/microsoft/azure/management/network/PacketCapture.java

```diff
@@ -0,0 +1,190 @@
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for
+ * license information.
+ */
+package com.microsoft.azure.management.network;
+
+import com.microsoft.azure.management.apigeneration.Beta;
+import com.microsoft.azure.management.apigeneration.Fluent;
+import com.microsoft.azure.management.network.implementation.PacketCaptureResultInner;
+import com.microsoft.azure.management.resources.fluentcore.arm.models.HasId;
+import com.microsoft.azure.management.resources.fluentcore.arm.models.HasName;
+import com.microsoft.azure.management.resources.fluentcore.model.Creatable;
+import com.microsoft.azure.management.resources.fluentcore.model.HasInner;
+import com.microsoft.azure.management.resources.fluentcore.model.Indexable;
+import com.microsoft.azure.management.storage.StorageAccount;
+
+import java.util.List;
+
+/**
+ * Client-side representation of Packet capture object, associated with Network Watcher.
+ */
+@Fluent
+@Beta
+public interface PacketCapture extends
+        HasInner<PacketCaptureResultInner>,
+        HasName,
+        HasId,
+        Indexable {
+    /**
+     * Get the target value.
+     *
+     * @return the target value
+     */
+    String targetId();
+
+    /**
+     * Get the number of bytes captured per packet, the remaining bytes are truncated.
+     *
+     * @return the bytesToCapturePerPacket value
+     */
+    Integer bytesToCapturePerPacket();
+
+    /**
+     * Get the maximum size of the capture output.
+     *
+     * @return the totalBytesPerSession value
+     */
+    Integer totalBytesPerSession();
+
+    /**
+     * Get the maximum duration of the capture session in seconds.
+     *
+     * @return the timeLimitInSeconds value
+     */
+    Integer timeLimitInSeconds();
+
+    /**
+     * Get the storageLocation value.
+     *
+     * @return the storageLocation value
+     */
+    PacketCaptureStorageLocation storageLocation();
+
+    /**
+     * Get the filters value.
+     *
+     * @return the filters value
+     */
+    List<PacketCaptureFilter> filters();
+
+    /**
+     * Get the provisioning state of the packet capture session. Possible values
+     * include: 'Succeeded', 'Updating', 'Deleting', 'Failed'.
+     *
+     * @return the provisioningState value
+     */
+    ProvisioningState provisioningState();
+
+    /**
+     * Stops a specified packet capture session.
+     */
+    void stop();
```

Comment:
also expose `Completable stopAsync()`. For all actions both async and sync should be exposed